src/llama-impl.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#define LLAMA_API_INTERNAL
+#include "llama.h"
+
+#ifdef __has_include
+    #if __has_include(<unistd.h>)
+        #include <unistd.h>
+        #if defined(_POSIX_MAPPED_FILES)
+            #include <sys/mman.h>
+            #include <fcntl.h>
+        #endif
+        #if defined(_POSIX_MEMLOCK_RANGE)
+            #include <sys/resource.h>
+        #endif
+    #endif
+#endif
+
+// bump if necessary
+#define LLAMA_MAX_NODES   8192
+#define LLAMA_MAX_LAYERS  512
```

Comment:
Did you mean to bump `LLAMA_MAX_LAYERS` from 256 to 512 here? 

I can't see how it would hurt anything, but I'm mainly asking out of paranoia re: functionality changes in PRs that are focused on refactoring.

Reply:
Yes, I actually (incorrectly) thought that https://github.com/ggerganov/llama.cpp/pull/8530 has been merged and now I noticed I have forgotten to merge it. So I just replicated the change here assuming it was already on `master`

Reply:
Aah, I hadn't seen that PR -- that all makes sense.

Thanks! :)